platforms/cuda/src/kernels/baoab.cu

```diff
@@ -0,0 +1,127 @@
+enum {VelScale, NoiseScale};
+
+/**
+ * Perform the first step of BAOAB integration.
```

Comment:
Would it make sense to provide a little bit of documentation about how you split the BAOAB update equations up among kernel calls? It looks like you fused the operations for a few equations here, and it would be good to tie specific equations in the publication to your logic in the implementation for future maintainability or development of variants (e.g. g-BAOAB, MTS versions, etc.).

Reply:
It pretty much just goes through the steps in order, nothing unusual at all.  The breaks between kernels just correspond to the places where it applies constraints or computes forces.  All the other integrators have their kernels split in exactly the same way.

Reply:
Can we at least minimally document that in the code? In general, it's great if the code is more readable and maintainable by everyone---if it's not obvious what is going on, a few extra comment lines on how you are breaking up the update equations into kernels is super helpful.

Reply:
Thanks! This is helpful!